From 07ce189d659a961780af152e2698af9d68660357 Mon Sep 17 00:00:00 2001
From: Moiz Sonasath <m-sonasath@ti.com>
Date: Sun, 13 Mar 2011 15:47:08 -0500
Subject: USB: MUSB: Move the c-state constarint calls to musb work queue

This patch moves the c-state constraint calls
"omap_pm_set_max_mpu_wakeup_lat" from musb notifier to the
musb work queue, scheduled in the musb interrupt handler.

Change-Id: Ic4d24b15989bb831761c666d9feca9c1d979b967
Signed-off-by: Moiz Sonasath <m-sonasath@ti.com>
---
 drivers/usb/musb/musb_core.c |    4 ++++
 drivers/usb/musb/omap2430.c  |    4 ----
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index d89e05f..ac47db4 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -1883,11 +1883,15 @@ static void musb_irq_work(struct work_struct *data)
 			plat->set_min_bus_tput(musb->controller,
 				OCP_INITIATOR_AGENT, (200*1000*4));
 	} else if (musb->xceiv->event == USB_EVENT_ID) {
+		/* Hold a c-state constarint */
+		omap_pm_set_max_mpu_wakeup_lat(&plat->musb_qos_request, 4000);
 		/* Hold a L3 constarint for better throughput */
 		if (plat->set_min_bus_tput)
 			plat->set_min_bus_tput(musb->controller,
 				OCP_INITIATOR_AGENT, (200*1000*4));
 	} else if (musb->xceiv->event == USB_EVENT_NONE) {
+		/* Release c-state constraint */
+		omap_pm_set_max_mpu_wakeup_lat(&plat->musb_qos_request, -1);
 		/* Release L3 constraint on detach*/
 		if (plat->set_min_bus_tput)
 			plat->set_min_bus_tput(musb->controller,
diff --git a/drivers/usb/musb/omap2430.c b/drivers/usb/musb/omap2430.c
index 64412ff..4e560fd 100644
--- a/drivers/usb/musb/omap2430.c
+++ b/drivers/usb/musb/omap2430.c
@@ -82,8 +82,6 @@ int musb_notifier_call(struct notifier_block *nb,
 		 * smart standby mode.
 		 */
 
-		omap_pm_set_max_mpu_wakeup_lat(&pdata->musb_qos_request, 4000);
-
 		musb_writel(musb->mregs, OTG_FORCESTDBY, 0);
 		val = musb_readl(musb->mregs, OTG_SYSCONFIG);
 		if (cpu_is_omap44xx())
@@ -156,8 +154,6 @@ int musb_notifier_call(struct notifier_block *nb,
 		val |= FORCEIDLE | FORCESTDBY;
 		musb_writel(musb->mregs, OTG_SYSCONFIG, val);
 
-		omap_pm_set_max_mpu_wakeup_lat(&pdata->musb_qos_request, -1);
-
 		val = __raw_readl(phymux_base +
 				USBA0_OTG_CE_PAD1_USBA0_OTG_DP);
 
-- 
1.6.0.4

